src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -149,9 +161,17 @@
 
 (schema/defn ^:always-validate get-jvm-metrics :- JvmMetricsV1
   []
-  (let [runtime-bean (ManagementFactory/getRuntimeMXBean)]
+  (let [runtime-bean (ManagementFactory/getRuntimeMXBean)
+        gc-beans (jmx/mbean-names "java.lang:name=*,type=GarbageCollector")]
     {:heap-memory (jmx/read "java.lang:type=Memory" :HeapMemoryUsage)
      :non-heap-memory (jmx/read "java.lang:type=Memory" :NonHeapMemoryUsage)
+     :file-descriptors (clojure.set/rename-keys
+                        (jmx/read "java.lang:type=OperatingSystem" [:OpenFileDescriptorCount :MaxFileDescriptorCount])
+                        {:OpenFileDescriptorCount :used :MaxFileDescriptorCount :max})
+     :gc-stats (map #(conj {:name (.getCanonicalName %)}
+                           (clojure.set/rename-keys (jmx/read % [:CollectionCount :CollectionTime])
+                                                    {:CollectionCount :count :CollectionTime :total-time-ms}))
+                    gc-beans)
```

Comment:
I need to pull this down and run it to confirm, but if I'm reading this correctly, the value for the `:gc-stats` entry in the status results will be a sequence/list, as opposed to a map/object.  That means that people would need to write code to iterate over that list (or assume they can call `first` on it) in order to consume this data.  Would prefer to see if we can figure out how to get this data converted into a more predictable, deterministic map structure.